cmd/kube-apiserver/app/server.go

```diff
@@ -82,13 +82,7 @@ cluster's shared state through which all other components interact.`,
 func Run(s *options.APIServer) error {
 	genericvalidation.VerifyEtcdServersList(s.ServerRunOptions)
 	genericapiserver.DefaultAndValidateRunOptions(s.ServerRunOptions)
-	genericConfig := genericapiserver.NewConfig(). // create the new config
-							ApplyOptions(s.ServerRunOptions). // apply the options selected
-							Complete()                        // set default values based on the known values
-
-	if err := genericConfig.MaybeGenerateServingCerts(); err != nil {
```

Comment:
An alternative idea was to create the GenericApiServer object completely independently from the Master object and pass a reference as a Master Config field (if needed). Then we have full control over the order. Moreover, we don't link Master and GenericApiServer unnecessarily.